packaging/docker/bin/entrypoint.sh

```diff
@@ -53,15 +53,18 @@ else
         $@
         exit $?
     fi
-    while true; do
-        es=$(taos -h $FIRST_EP_HOST -P $FIRST_EP_PORT --check)
-        echo ${es}
-        if [ "${es%%:*}" -eq 2 ]; then
-            echo "execute create dnode"
-            taos -h $FIRST_EP_HOST -P $FIRST_EP_PORT -s "create dnode \"$FQDN:$SERVER_PORT\";"
-            break
-        fi
-        sleep 1s
-    done
+    # check if dnode has been created
+    if [ ! -f "$DATA_DIR/dnode/dnode.json" ]; then
```

Comment:
Actually, the check should be fixed in [Lline 46](https://github.com/taosdata/TDengine/blob/6f53e8ed764a82ecfa92787838bfed0b9128b38b/packaging/docker/bin/entrypoint.sh#L46), since TDengine 3.0 has changed its data structure - its a mistake though.

Reply:
I agree with your suggestion and have changed it.